flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -318,17 +319,10 @@ public void testStateBackendClosingOnFailure() throws Exception {
 	@Test
 	public void testCancellationNotBlockedOnLock() throws Exception {
 		syncLatch = new OneShotLatch();
-
-		StreamConfig cfg = new StreamConfig(new Configuration());
-		Task task = createTask(CancelLockingTask.class, cfg, new Configuration());
-
-		// start the task and wait until it runs
-		// execution state RUNNING is not enough, we need to wait until the stream task's run() method
-		// is entered
+		task = createTask(CancelLockingTask.class, new StreamConfig(new Configuration()), new Configuration());
 		task.startTaskThread();
-		syncLatch.await();
 
-		// cancel the execution - this should lead to smooth shutdown
+		syncLatch.await(); // wait for the main loop to start
 		task.cancelExecution();
 		task.getExecutingThread().join();
 
```

Comment:
Add `task = null`.

In general, not a big fan of static variables in tests, but I don't see how we can improve that now.

Reply:
Improving the tests is not the scope of this PR.

Reply:
Improving existing tests is out of scope of this PR.

Reply:
Yes, but in this case you made a local variable global, so make sure to also reset it.

Reply:
You're right, addressed it.